### PR TITLE
feat: added puppeteer loader node

### PR DESCRIPTION
**Pull request by AidanTilgner**
I updated the `components` package to add the puppeteer integration described in [this issue](https://github.com/FlowiseAI/Flowise/issues/311). This is my first PR here, so let me know if I need to change anything/everything.

Screenshot 2023-06-13 at 1 18 00 PM

- [x] Add Puppeteer Loader node
- [x] Puppeteer Loader node should be able to scrape data from webpages
- [x] Build successful 

> **Note:** Puppeteer is updating their default `headless` mode, and therefore a warning is displayed and the option to update the `headless` proeprty to `headless: "new"` is offered. However, it seems that the decision to update to `headless: "new"` is controlled by the `PuppeteerWebBaseLoader` langchain loader. I don't think this is an issue, as the new headless mode will become the default soon, but I figured I'd mention it.

### CODE REVIEW

**Comment by AidanTilgner** [comment]
It seems that there are some merge conflicts now since the new emergency changes were made. I will update this tomorrow.

**Comment by AidanTilgner** [comment]
Screenshot 2023-06-14 at 10 52 21 AM
ert

I ran the build and start command according to the [contributing guidelines](https://github.com/FlowiseAI/Flowise/blob/main/CONTRIBUTING.md). I also created the above chain for testing purposes. I confirmed that the "Puppeteer Web Scraper" node was scraping documents successfully. Then I used the "Pinecone Upsert Document" node to confirm that they were being scraped and then pulled into memory by the language model. Everything works the same as the [cherio reference code](https://github.com/FlowiseAI/Flowise/blob/main/packages/components/nodes/documentloaders/Cheerio/Cheerio.ts).

File: packages/components/package.json [hunk 1]
```diff
@@ -38,9 +38,9 @@
         "moment": "^2.29.3",
         "node-fetch": "^2.6.11",
         "pdf-parse": "^1.1.1",
+        "puppeteer": "^20.7.1",
         "weaviate-ts-client": "^1.1.0",
-        "ws": "^8.9.0",
-        "html-to-text": "^9.0.5"
```
Comment (HenryHengZJ) [inline packages/components/package.json:43]: accidentally removed `html-to-text`?

Comment (AidanTilgner) [reply packages/components/package.json:43]: Yeah I think I accidentally deleted it when resolving the merge conflict

Comment (AidanTilgner) [reply packages/components/package.json:43]: Added a fix

**Review by HenryHengZJ — APPROVED** [review]
thanks @AidanTilgner! lgtm

**Pull request merged**